File: python/calculator.py
import math

def perform_operation(choice, num1, num2):
    if choice == 1:
        return num1 + num2
    elif choice == 2:
        return num1 - num2
    elif choice == 3:
        return num1 * num2
    elif choice == 4:
        if num2 != 0:
            return num1 / num2
        else:
            print("Error: Division by zero")
            return None
    elif choice == 6:
        return num1 ** num2
    elif choice == 7:
        if num1 >= 0:
            return math.sqrt(num1)
        else:
            print("Error: Cannot calculate square root of a negative number")
            return None
    elif choice == 8:
        return num1 % num2
    elif choice == 9:
        return math.sin(math.radians(num1))
    elif choice == 10:
        return math.cos(math.radians(num1))
    elif choice == 11:
        return math.tan(math.radians(num1))
    else:
        print("Invalid choice")
        return None

def main():
    while True:
        print("1. Addition")
        print("2. Subtraction")
        print("3. Multiplication")
        print("4. Division")
        print("5. Quit")
        print("6. Exponentiation")
        print("7. Square Root")
        print("8. Modulus")
        print("9. Sine")
        print("10. Cosine")
        print("11. Tangent")

        choice = int(input("Enter your choice (1/2/3/4/5/6/7/8/9/10/11): "))

        if choice == 5:
            print("Exiting the calculator.")
            break

        num1 = float(input("Enter number: "))

        if choice not in [7, 9, 10, 11]:
            num2 = float(input("Enter number: "))
        else:
            num2 = 0

        result = perform_operation(choice, num1, num2)

        if result is not None:
            print("Result:", result)

        print()

if __name__ == "_main_":
    main()

File: calculator/calculator.php
<?php
function performOperation($choice, $num1, $num2) {
    switch ($choice) {
        case 1:
            return $num1 + $num2;
        case 2:
            return $num1 - $num2;
        case 3:
            return $num1 * $num2;
        case 4:
            if ($num2 != 0) {
                return $num1 / $num2;
            } else {
                echo "Error: Division by zero\n";
                return PHP_INT_MIN;
            }
        case 6:
            return pow($num1, $num2);
        case 7:
            if ($num1 >= 0) {
                return sqrt($num1);
            } else {
                echo "Error: Cannot calculate square root of a negative number\n";
                return PHP_INT_MIN;
            }
        case 8:
            return $num1 % $num2;
        case 9:
            return sin(deg2rad($num1));
        case 10:
            return cos(deg2rad($num1));
        case 11:
            return tan(deg2rad($num1));
        default:
            echo "Invalid choice\n";
            return PHP_INT_MIN;
    }
}

while (true) {
    echo "1. Addition\n";
    echo "2. Subtraction\n";
    echo "3. Multiplication\n";
    echo "4. Division\n";
    echo "5. Quit\n";
    echo "6. Exponentiation\n";
    echo "7. Square Root\n";
    echo "8. Modulus\n";
    echo "9. Sine\n";
    echo "10. Cosine\n";
    echo "11. Tangent\n";

    echo "Enter your choice (1/2/3/4/5/6/7/8/9/10/11): ";
    $choice = intval(fgets(STDIN));

    if ($choice == 5) {
        echo "Exiting the calculator.\n";
        break;
    }

    echo "Enter number: ";
    $num1 = floatval(fgets(STDIN));

    $num2 = 0; // Initialize num2

    if ($choice != 7 && $choice != 9 && $choice != 10 && $choice != 11) {
        echo "Enter number: ";
        $num2 = floatval(fgets(STDIN));
    }

    $result = performOperation($choice, $num1, $num2);

    if ($result != PHP_INT_MIN) {
        echo "Result: " . $result . "\n";
    }

    echo "\n";
}
?>